### PR TITLE
adding back cudnn 5.1 support

### LINKED ISSUE #1607: Bring back cuDNN 5.1 support
This has disappeared since we're not testing for it, but it might be easy to bring back.

@slayton58 @yangqing Worth it?

### PR DESCRIPTION
**Pull request by Yangqing**
Manually verified via docker.

Closes #1607 .

cc #1343 #1302 #1584

### CODE REVIEW

**Comment by Yangqing** [comment]
For the record, cudnn 5.1 will only be supported in a best-effort basis so one is strongly encouraged to update to cudnn 6 and above.

File: caffe2/operators/pool_op_cudnn.cu [hunk 1]
```diff
@@ -493,7 +507,7 @@ class CuDNNPoolGradientOp : public ConvPoolOpBase<CUDAContext> {
 
   // Input: X, Y, dY
   // Output: dX
-  INPUT_TAGS(IN, OUT, OUT_GRAD);
+  // INPUT_TAGS(IN, OUT, OUT_GRAD);
```
Comment (pietern) [inline caffe2/operators/pool_op_cudnn.cu]: comment can go?

**Comment by pietern** [comment]
The bitonic sort thing is also needed for this one?

I'm adding a Jenkins job to test this.

File: caffe2/operators/pool_op_cudnn.cu (hunk 1 above)
Comment (Yangqing) [reply caffe2/operators/pool_op_cudnn.cu]: Ah yup :)

**Comment by Yangqing** [comment]
The bitonic sort is basically needed for windows (which is a separate topic), but yeah, it helps clean up code clutter so I included it here :)

**Comment by pietern** [comment]
Build: https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda8.0-cudnn5-ubuntu16.04-trigger-test/2

If it passed and this PR is merged then I'll add it to the default set.

**Comment by pietern** [comment]
```
00:01:11.875 --     CUDA version        : 8.0
00:01:11.875 --     CuDNN version       : 5.1.10
```

**Comment by pietern** [comment]
@Yangqing There's a couple of test failures, see [the test build](https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda8.0-cudnn5-ubuntu16.04-test/1/)

**Comment by pietern** [comment]
Triggered rebuild [here](https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda8.0-cudnn5-ubuntu16.04-trigger-test/3/).

It is not part of the default set of builds yet.

**Comment by pietern** [comment]
Triggered rebuild [here](https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda8.0-cudnn5-ubuntu16.04-trigger-test/4/).

**Comment by Yangqing** [comment]
cudnn 5 passed here: https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda8.0-cudnn5-ubuntu16.04-trigger-test/

**Pull request closed without merging**